Feat/microloan

shirleylycode8 opened the pull request.

shirleylycode8 committed:
add src directory

shirleylycode8 committed:
add payment functionality

shirleylycode8 committed:
Address code review comment

shirleylycode8 committed:
Implement loan functionality

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/BankLender.java:
```diff
@@ -0,0 +1,26 @@
+
+import java.util.List;
+
+public class BankLender {
```
This class doesn't seem to be used at all, is this intended for future development?

shirleylycode8 replied:
deleted.

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/Customer.java:
```diff
@@ -6,6 +6,24 @@
+
+	public void setAccounts(ArrayList<Account> accounts) {
+		this.accounts = accounts;
+	}
```
The getAccounts() and setAccounts() methods aren't being used. It's probably fine to keep them in, as they might be useful later, but I thought I'd mention it just in case this wasn't intentional.

shirleylycode8 replied:
Ok noted. Left it.

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/NewBank.java:
```diff
@@ -14,9 +14,10 @@ private NewBank() {
-		//NewBank bank = new NewBank();
 
 		Customer bhagy = new Customer();
+		bhagy.setLender("bhagy lender");
```
The LENDER is interpreted as just one word with no spaces in the "REGISTERLENDER" and "BORROWMICROLOAN" features because the input string is split by " ". So if we are declaring any lenders from the start, they will have to have no spaces too, otherwise the lender name will be split.

shirleylycode8 replied:
addressed.

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/MicroLoan.java:
```diff
@@ -0,0 +1,35 @@
+    public void setCurrentIncome(double currentIncome) {
+        this.currentIncome = currentIncome;
+    }
+
```
The 'getters' in this class are not used. It's probably fine to keep them in, as they might be useful later, but I thought I'd mention it just in case this wasn't intentional.

Edit: Just realised it's not very clear that I'm referring to the 'getters' in the MicroLoan class, not the line that I've left this comment on!

shirleylycode8 replied:
noted.

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/NewBank.java:
```diff
@@ -46,13 +47,30 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
 	}
 
-	// commands from the NewBank customer are processed in this method
+	// commands from the NewBank cusbtomer are processed in this method
```
Small typo.

shirleylycode8 replied:
corrected

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/NewBank.java:
```diff
@@ -46,13 +47,30 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
+			}
+			if (request.startsWith("SHOWMICROLENDERS")){
+				return showMicroLenders(); // +1 to remove the leading space
+			}
```
SHOWMICROLENDERS feature could be added to the protocol.txt

shirleylycode8 replied:
Done

ActuarialAlchemist commented on New Bank/NewBank/newbank/server/NewBank.java:
```diff
@@ -46,13 +47,30 @@ public synchronized CustomerID checkLogInDetails(String userName, String passwor
 				return openAccount(customer, request.substring(request.indexOf(" ") + 1)); // +1 to remove the leading space
 			}
+			if (request.startsWith("REGISTERLENDER")){
+				if(input.length < 1) { // return fail if not enough information is provided
```
I think it should be <2 instead of <1

shirleylycode8 replied:
done

ActuarialAlchemist reviewed (changes requested):
I've made a few comments below, but the main question I had is about the 'lender' aspect of this.

What is a Lender meant to represent in the context of this feature? I noticed it was part of the Customer class, so is it the Lender for that specific Customer?

ActuarialAlchemist commented:
> I've made a few comments below, but the main question I had is about the 'lender' aspect of this.
> 
> What is a Lender meant to represent in the context of this feature? I noticed it was part of the Customer class, so is it the Lender for that specific Customer?

ActuarialAlchemist commented:
Sorry, accidentally closed the pull request. Reopening it.

shirleylycode8 commented:
> > the main question I had is about the 'lender' aspect of this.
> > What is a Lender meant to represent in the context of this feature? I noticed it was part of the Customer class, so is it the Lender for that specific Customer?

The customer also has the ability to lend money.

shirleylycode8 committed:
Update protocol.txt

shirleylycode8 committed:
Update NewBank.java

shirleylycode8 committed:
Delete BankLender.java

hithesh98 committed:
Fixmicroloan (#20)

* retrive data from csv file, MOVE function not updated

* edit addAccount, writting to csv file not functional

* Changes for SIGNUP and LOGIN functionality

* store data outside runtime session. Does not work for MOVE and PAY functions

* Addition to signup:
if(customers.containsKey(name)) { return "FAIL"; }

* Implemented LOGOFF function

* Revert "Implemented LOGOFF function"

This reverts commit 230315e4

* I added LOGOFF function

* Remove redundant string un and method getUn

* Logoff removed idea files (#13)

* Update NewBank.java

* Moved login into separate method.

* Corrected 'responce' spelling mistake

Co-authored-by: hithesh98 <42919655+hithesh98@users.noreply.github.com>

* split csv into two. One handles user info other handles transaction info

* Move function updates outside runtime session

* I fix the bug if user login or password is wrong (#15)

* I fix the bug if user login or password is wrong

* Added punctuation to msg

Co-authored-by: hithesh98 <42919655+hithesh98@users.noreply.github.com>

* Signup changed to write to the next line on CSV files and includes own password option in the protocol.

Sample data I have tested on saved in ledger.csv and users.csv.

* Update Requirements

I added and put in the right format functional requirements apart the ones linked with microloans.

Co-authored-by: James McDermott <76912128+jm3138@users.noreply.github.com>
Co-authored-by: virgul13 <micprzyb@gmail.com>
Co-authored-by: Authiel <calummcdonald54@gmail.com>
Co-authored-by: Calum <68914772+Authiel@users.noreply.github.com>
Co-authored-by: virgul13 <42522421+virgul13@users.noreply.github.com>

hithesh98 committed:
Merge branch 'main' into feat/microloan
